src/main/java/com/exasol/swqa/maven/SummarizerMojo.java

```diff
@@ -0,0 +1,120 @@
+package com.exasol.swqa.maven;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * The SummarizerMojo class is a Maven plugin goal that performs quality metric summarization.
+ * <p>
+ * It implements the Mojo interface and extends the AbstractMojo class. The goal "summarize" and runs in the default
+ * phase "VERIFY".
```

Comment:
```suggestion
 * The goal is called "summarize" and runs in the default phase "VERIFY".
```
Not sure if it's helpful to state which class this extends, because you can see this some lines below. Also goal name and phase are close by.